karapace/backup/api.py

```diff
@@ -424,10 +425,12 @@ def restore_backup(
 
         for instruction in backend.read(backup_location, topic_name):
             if isinstance(instruction, RestoreTopicLegacy):
-                _handle_restore_topic_legacy(instruction, config)
+                if not skip_topic_creation:
+                    _handle_restore_topic_legacy(instruction, config)
                 producer = stack.enter_context(_producer(config, instruction.topic_name))
             elif isinstance(instruction, RestoreTopic):
-                _handle_restore_topic(instruction, config)
+                if not skip_topic_creation:
+                    _handle_restore_topic(instruction, config)
```

Comment:
suggestion (non-blocking): This increases cyclic complexity, and is sort of orthogonal to what this function should do (delegate to the correct handler), and might be preferable to do as early return checks within the handler functions instead.